docs/getting-started.md

```diff
@@ -31,9 +31,25 @@ There's also an extra package: `@ecl-twig/{ec,eu}-components` which is used to a
 
 Each component comes with a story `*.story.js` which is consumed by [Storybook](https://storybook.js.org/). In this story file, you import the main Twig template and you pass data to it. That's it!
 
-You can use knobs to make the showcase more dynamic and interactive.
+In ecl-twig the storybook knobs are extensively used to accurately represent the full input data with the aim of offering the possibility to customise the component and retrieve the correct object to the passed to the component in the `notes` tab.
+Their label is supposed to be defined the same way the property is accessible in the expected input data.
+Ex: `link.icon.path`
 
-More features will come in the near future...
+Knobs in ecl-twig are grouped together to simplify their usage and offer additional information, like whether the property controlled by the knob is mandatory or optional, for instance, the available groups are _mandatory_, _optional_, _use cases_, _status_
+In ecl-twig buttons knobs are used to defined different use cases the component might have, when blocks of information are optional and can be toggled on and off in a component demo.
+In many cases (like icons, path to the logo...) there is no real advantage for the user in editing the value used in the demo, therefore we only offer the possibility of unsetting it, mainly for testing the compliance checks we perform.
+
+### ECL Compliance
+
+The ecl-twig library comes with a special component named `ecl-compliance`, which defines a parent template for the ecl compliance checks to be performed, it also contains all the children templates for the checks applied to the single components.
+Each of our templates should have a corresponding child template defined so to make a compliance check available through our storybook demo and to all the consumers of the library thanks to the ecl-compliance package and a parameter applied to the component instantiation: `_ecl_compliance_`.
+
+An additional parameter `_ecl_compliance_inner_check_` is available for identifying the inner checks run by a component on all its included templates from the overall compliance performed on a component. This offers us the possibility of differentiating the output of the report. Mind that for this parameter to work also the `icon_path` needs to be available in order to render an icon for the success or the failure of the checks.
+
+The file name of the child template is expected to follow this convention:
+_ecl-compliance-{nameOfTheComponent}.html-twig_
```

Comment:
In a similar way a react template is created for demo during work done on a component in ECL and this is seen as overcomplication, ecl-twig is growing in terms of information encoded in components' templates. Tricky and proving that any project growing adds code to cover some purpose, but don't be surprised if someone tells you that complience checker templates in addition to components templates is over-complication.

The tricky part is not so much about writing loads of code with checks, but maintaining it. Especially having it in twig language which is not the most interoperable.

Reply:
I find the comparison with the situation in ECL you mentioned totally arbitrary, @kalinchernev, not following it, then.
it's not really surprising indeed to deal with some additional code in implementing a new functionality, what might be surprising is maybe to consider this an overhead in this case.
 
In my experience, for instance, the real overhead in working on this library was the total lack of confidence the developers had about the "quality" of the components they were implementing, because there was not much there to help them.
Now we have many pillars where to base our confidence on, instead, and yes, each of them required work and code, also extensively adding knobs to all the stories was an overhead, but the additional value of it justifies it, if the users are now capable of seeing and editing the whole data structure and retrieve the right include code after they customised it.

In this case the added value is both for us, as developers and for the final users, some twig logic in a template doesn't sound as an extraordinary effort on our side, it's a twig templates library, nothing really looks tricky or complicated, to be honest.
About the "maintenance", this would only be an issue if the existing components would run into multiple iterations, which doesn't seem the case in a library that should be close to its final state and anyway i hardly see how this would be "critical", it's just an additional task for the developer, but it's for the good.